core/genesis.go

```diff
@@ -475,11 +475,15 @@ func (g *Genesis) ToBlock() *types.Block {
 		Coinbase:   g.Coinbase,
 		Root:       root,
 	}
-	if g.GasLimit == 0 {
-		head.GasLimit = params.GenesisGasLimit
-	}
-	if g.Difficulty == nil && g.Mixhash == (common.Hash{}) {
-		head.Difficulty = params.GenesisDifficulty
+	// Disable setting default gas limit and difficulty in cel2 context.
+	// Because the celo genesis lacked gas limit and difficulty.
+	if g.Config.Cel2Time == nil {
```

Comment:
Can we modify this check to something similar to the one [here](https://github.com/celo-org/op-geth/pull/143/files#diff-1272116434e5549f278a940767af52ed2d1abc91c44fa04ec48d8835a3774036R488) and check if Cel2 was activated in block 0 (which happens only in tests) and set values appropriately?

Reply:
Yeah, I think this will work!